### PR TITLE
Replace deprecated primitive-to-object conversions

### PR DESCRIPTION
**Pull request by edewata**
The constructors used to convert primitives into the corresponding objects were deprecated in Java 9 so they have been replaced with autoboxing.

https://bugs.openjdk.java.net/browse/JDK-8176335

### CODE REVIEW

File: java-sdk/ldapjdk/src/main/java/netscape/ldap/LDAPConnThread.java [hunk 1]
```diff
@@ -721,7 +721,7 @@ private synchronized void cacheSearchResult (LDAPSearchListener l, LDAPMessage m
             v = m_messages.get(messageID);
             if (v == null) {
                 m_messages.put(messageID, v = new Vector<Object>());
-                v.addElement(new Long(0));
+                v.addElement(0);
```
Comment (ckelleyRH) [inline java-sdk/ldapjdk/src/main/java/netscape/ldap/LDAPConnThread.java]: This needs to be `0L` or we have changed type to `Integer` and will get `ClassCastException` on line 728.

File: java-sdk/ldapjdk/src/main/java/netscape/ldap/LDAPConnThread.java [hunk 2]
```diff
@@ -778,7 +778,7 @@ private synchronized void cacheSearchResult (LDAPSearchListener l, LDAPMessage m
                 // server
                 if (v == null) {
                     v = new Vector<Object>();
-                    v.addElement(new Long(0));
```
Comment (ckelleyRH) [inline java-sdk/ldapjdk/src/main/java/netscape/ldap/LDAPConnThread.java:781]: `0L`

File: java-sdk/ietfldap/org/ietf/ldap/LDAPConnThread.java [hunk 3]
```diff
@@ -605,7 +604,7 @@ private synchronized void cacheSearchResult( LDAPSearchQueue l,
             v = _messages.get( messageID );
             if ( v == null ) {
                 _messages.put( messageID, v = new Vector<>() );
-                v.addElement( new Long(0) );
+                v.addElement( 0 );
```
Comment (ckelleyRH) [inline java-sdk/ietfldap/org/ietf/ldap/LDAPConnThread.java]: `0L`

File: java-sdk/ietfldap/org/ietf/ldap/LDAPConnThread.java [hunk 4]
```diff
@@ -662,7 +661,7 @@ private synchronized void cacheSearchResult( LDAPSearchQueue l,
                 // server
                 if ( v == null ) {
                     v = new Vector<>();
-                    v.addElement(new Long(0));
+                    v.addElement(0);
```
Comment (ckelleyRH) [inline java-sdk/ietfldap/org/ietf/ldap/LDAPConnThread.java]: `0L`

**Review by ckelleyRH — CHANGES_REQUESTED** [review]
Mostly fine but some values that were `Long` were converted to `Integer`. The compiler is fine with it because those values are getting added to a `Vector<Object>` but then at runtime when you try to cast that `Integer` element to a `Long` it blows up throwing `ClassCastException` (I verified it with a simple test).

**Comment by edewata** [comment]
@ckelleyRH Great catch! Please see the updated PR. Thanks!

**Review by ckelleyRH — APPROVED** [review]
Looks good now, thanks!

**Comment by edewata** [comment]
@ckelleyRH Thanks!

**Pull request merged**